manifests/install.pp

```diff
@@ -1,33 +1,30 @@
 # Installs packages required to utilize winbind for joining Active Directory
-class winbind::install (
-  $enable_sharing = $::winbind::enable_sharing,
-  $package_ensure = $::winbind::package_ensure,
-  ) {
+class winbind::install inherits winbind {
   case $::osfamily {
     'RedHat' : {
       case $::operatingsystemmajrelease {
         '5'     : {
           package { 'samba3x-winbind':
-            ensure => $package_ensure,
+            ensure => $winbind::package_ensure,
           }
 
-          if ($enable_sharing) {
+          if ($winbind::enable_sharing) {
             package { 'samba3x':
-              ensure => $package_ensure,
+              ensure => $winbind::package_ensure,
             }
           }
         }
 
         default : {
-          $packages = ['samba-winbind-clients', 'oddjob-mkhomedir']
+          $_packages = ['samba-winbind-clients', 'oddjob-mkhomedir']
```

Comment:
Let me know if I am not thinking about this correctly, but that `$packages` variable was never in the params class, just local to the install class, so following the naming [docs](https://docs.puppet.com/puppet/5.0/lang_variables.html#naming), since that variable will only be accessed from its local scope I added the underscore.

However, this could be refactored by moving all the package names to a variable in their respective hiera file, which I think would be better. What do you think?